ecocheck.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from infra.system.helpers.utils import getScriptDir
+from infra.system.helpers.schema_validator import SchemaValidator
+import time
+import datetime
+from gofed.lib.pkgdb2client import get_packages
+from gofed.lib.bitbucketclient import get_commits_bitbucket
+from gofed.lib.githubclient import get_commits_github
+from gofed.lib.kojiclient import KojiClient
+
+
+SUPPORTED_REPOSIRORIES = ['github.com', 'bitbucket.org']
+
+
+def repository_info(clone_url):
+	if not clone_url:
+		return []
+	data = clone_url.split('/')
+	if data[2] in SUPPORTED_REPOSIRORIES:
+		repository = {
+			'host': data[2],
+			'owner': data[3],
+			'package': data[4].split('.')[0]
+		}
+	else:
+		logging.error('Unsupported upstream host: %s' % data[2])
+		return []
+	return repository
+
+
+class EcoCheck:
+	def __init__(self):
+		self.input_validated = False
+		self.new_packages = []
+		self.updated_upstream_repos = []
+		self.new_koji_builds = []
+		self.packages = []
+		self.no_prefix_packages = []
+		self.date = ""
+
+	def _validateInput(self, data):
+		validator = SchemaValidator()
+		schema = "%s/ecocheck_input.json" % getScriptDir(__file__)
+		self.input_validated = validator.validateFromFile(schema, data)
+		return self.input_validated
+
+	def setData(self, data):
+		self.input_validated = False
+		if not self._validateInput(data):
+			return []
+		self.date = data['last_change_date']
+		self.no_prefix_packages = data['no_prefix_packages']
+
+		return True
+
+	def _generateGolangProjectUpdatedPackages(self):
+		data = {}
+		data['last_change_date'] = self.date
+		data['new_packages'] = self.new_packages
+		data['updated_upstream_repositories'] = self.updated_upstream_repos
+		data['new_koji_builds'] = self.new_koji_builds
+
+		return data
+
+	def getData(self):
+		if not self.input_validated:
+			return []
+
+		data = self._generateGolangProjectUpdatedPackages()
+		validator = SchemaValidator()
+		schema = "%s/updated-packages.json" % getScriptDir(__file__)
+		if not validator.validateFromFile(schema, data):
+			logging.error('golang-project-updated-packages is not valid')
+			return {}
+
+		return data
+
+	def execute(self):
+		self.no_prefix_packages.append('golang*')
+		self.packages = []
+		for packagename in self.no_prefix_packages:
+			self.packages = get_packages(packagename)
+		file = open('unresolved_packages.txt', 'w')
+
+		for package in self.packages:
+			if package['upstream_url'] == "":
```

Comment:
``upstream_url`` picked from Pkgdb will not always be the correct one. At the same time its value depends on a values in package's spec file. This should be read from a database/local file. At this point there is no class providing a list of upstream_url a.k.a provider_prefix. It can be read from spec file from %provider_prefix macro. Still, the macro is here more for spec file validation and for tools working over spec files/rpms. Upstream repository url/provider url should be independent of what is in a spec file.

I would suggest to create a local json file with a list of packagename -> repositoryurl mapping based on [1]. And read this file instead of using ``upstream_url`` from pkgdb's package info.

[1] https://github.com/gofed/gofed/blob/master/data/golang.repos